### ISSUE TITLE
🐛 Bug: Phantom Upgrade

### ISSUE THREAD

**Issue by brandondrew** [nocodb/nocodb#9084]
### Please confirm that the bug report does ***not*** already exist

- [X] I confirm there is no existing issue for this bug.

### Steps to reproduce

The NocoDB UI claims an upgrade is available, however 

1. nothing shows up on the releases page for 0.251.3;
2. `brew upgrade nocodb` fails (see below);
3. there is no code change: https://github.com/nocodb/nocodb/compare/0.251.3...0.251.2;

```shell
==> Upgrading 1 outdated package:
nocodb/nocodb/nocodb 0.251.2 -> 0.251.3
==> Fetching nocodb/nocodb/nocodb
==> Downloading https://github.com/nocodb/nocodb/releases/download/0.251.3/nocodb.tar.gz
curl: (22) The requested URL returned error: 404                                                                                                                                               

Error: nocodb: Failed to download resource "nocodb"
Download failed: https://github.com/nocodb/nocodb/releases/download/0.251.3/nocodb.tar.gz
```

### Desired Behavior

If there is no upgrade available, the menu in the NocoDB UI should not claim there is one available.  Clearly it is using different criteria, such as possibly the presence of a tag.  But a tag without a release to download is useless.  Instead, everything should depend on the presence of an actual release being available for download.

### Project Details

Node: **v18.5.0**
Arch: **x64**
Platform: **darwin**
Docker: **false**
RootDB: **sqlite3**
PackageVersion: **0.251.2**

### Attachments

<img width="321" alt="image" src="https://github.com/user-attachments/assets/564b34e9-9fa2-40af-aa09-3bc399543f8f">
<img width="811" alt="image" src="https://github.com/user-attachments/assets/d8b68e88-8a7d-45ed-a6b7-84857016548b">
<img width="658" alt="image" src="https://github.com/user-attachments/assets/ad4dcfef-0728-4cf1-92fc-225ef75ebc18">

**Comment by o1lab** [comment]
There is an issue with releasing executables and brew - team are looking into it.

And the release details are available [here](https://github.com/nocodb/nocodb/releases/tag/0.251.3).

**Comment by brandondrew** [comment]
@o1lab take a look at my last 2 screenshots:

- there was nothing available for 0.251.3 on https://github.com/nocodb/nocodb/releases
- there was no download available for 0.251.3 on the tags page

I don't see any way this can possibly be a Homebrew problem.  If you've reported it to them as their problem, I think you're sending them on a wild goose chase.

Is there a delay between when you tag a branch and when you create a release?  (I assume _some_ delay is inevitable.)  Why not make the "Upgrade Available" button depend on a release being available, rather than a branch being tagged?

**Comment by o1lab** [comment]
@brandondrew : the release was in draft state when this issue was reported and since notified (thanks to you) we turned from draft to released. As indicated there were issues with binaries getting created.

And then the changes b/w tags you are looking at the changes from 0.251.3 to 0.251.2 https://github.com/nocodb/nocodb/compare/0.251.3...0.251.2

instead of 0.251.2 to 0.251.3 => https://github.com/nocodb/nocodb/compare/0.251.2...0.251.3
